ts/parts-map/HeatmapSeries.ts

```diff
@@ -514,7 +606,47 @@ seriesType<Highcharts.HeatmapSeriesOptions>(
 
         /* eslint-enable valid-jsdoc */
 
-    }, colorPointMixin)
+    },
+    colorPointMixin,
+    {
+        setState: function (
+            this: Highcharts.HeatmapPoint,
+            state?: string
+        ): void {
+            var point = this,
+                series = this.series,
+                points = series.points,
+                inactiveException = (series.options.states as any)
+                    .inactive.except as Highcharts
```

Comment:
This cast should not be necessary in the same way. Also try to write types on one line and if it is not possible, do
```
something{:| as} (
    The.Extremly.Long.Type.Name
)
```

Reply:
Ok, thanks!